### PR TITLE
[pallet_collective] Enforce prime is a valid member of collective in set_members extrinsic

### LINKED ISSUE #14353: [pallet_collective] set_members Extrinsic Allows Arbitrary 'prime' Argument
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Stack Exchange first?

- [X] This is not a support question.

### Motivation

The motivation behind filing this issue is to address problematic behaviour in the `pallet_collective` module of the FRAME pallets. The current implementation of the `set_members` extrinsic allows for an arbitrary value to be passed as the `prime` argument, despite its intended purpose of representing one of the members in the `new_members` argument. This behaviour deviates from the expected functionality. 

### Request

The request is to modify the `set_members` extrinsic in the `pallet_collective` module of the FRAME pallets to enforce the intended usage of the `prime` argument. 

### Solution

To address this issue, I propose that the `prime` argument should be restricted to accepting only one of the accounts listed in the `new_members` argument. This modification will ensure that the `set_members` extrinsic functions as intended, allowing for correct and secure updates to the collective membership. I would be happy to raise a PR to fix this issue.

### Are you willing to help with this request?

Yes!

### PR DESCRIPTION
**Pull request by ToufeeqP**
This fixes #14353

Polkadot address: 1WSnZDtb4FEufmwdQh2emEXdDCmwZ5pSoPfsgStQb3kQa9M

### CODE REVIEW

File: frame/collective/src/lib.rs [hunk 1]
```diff
@@ -416,6 +418,9 @@ pub mod pallet {
 					old.len(),
 				);
 			}
+			if let Some(p) = prime.clone() {
```
Comment (KiChjang) [inline frame/collective/src/lib.rs:421]:
```suggestion
			if let Some(p) = &prime {
```

File: frame/collective/src/lib.rs [hunk 2]
```diff
@@ -416,6 +418,9 @@ pub mod pallet {
...
+				ensure!(new_members.contains(&p), Error::<T, I>::PrimeAccountNotMember);
```
Comment (KiChjang) [inline frame/collective/src/lib.rs:422]:
```suggestion
				ensure!(new_members.contains(p), Error::<T, I>::PrimeAccountNotMember);
```

**Review by sam0x17 — APPROVED** [review]

**Review by juangirini — APPROVED** [review]

**Review by KiChjang — APPROVED** [review]

**Review by muharem — APPROVED** [review]

**Comment by muharem** [comment]
bot merge

**Comment by juangirini** [comment]
hi @ToufeeqP should you want get tipped with DOT/KSM, please add your address in the PR comment with this format:
[network] address: [address]
Eg: Kusama address: xxxxxxx

**Comment by ToufeeqP** [comment]
Thank you for asking.
Polkadot address: 1WSnZDtb4FEufmwdQh2emEXdDCmwZ5pSoPfsgStQb3kQa9M

**Pull request merged**